tt-train/CMakeLists.txt

```diff
@@ -0,0 +1,73 @@
+cmake_minimum_required(VERSION 3.16)
+cmake_policy(VERSION 3.16)
+include(cmake/compilers.cmake)
+
+if(DEFINED ENV{CMAKE_C_COMPILER} AND DEFINED ENV{CMAKE_CXX_COMPILER})
+    message(STATUS "Setting C and C++ compiler from environment variables")
+    set(CMAKE_C_COMPILER $ENV{CMAKE_C_COMPILER})
+    set(CMAKE_CXX_COMPILER $ENV{CMAKE_CXX_COMPILER})
+endif()
+
+if(CMAKE_CXX_COMPILER AND CMAKE_C_COMPILER)
+    message(STATUS "Using specifed C++ compiler: ${CMAKE_CXX_COMPILER}")
+    message(STATUS "Using specifed C compiler: ${CMAKE_C_COMPILER}")
+else()
+    message(STATUS "No C or C++ compiler specified, defaulting to Clang-17")
+    FIND_AND_SET_CLANG17()
+endif()
+
+project(ml-framework-cpp)
+CHECK_COMPILERS()
+
+if("$ENV{TT_METAL_HOME}" STREQUAL "")
+    message(FATAL_ERROR "TT_METAL_HOME is not set")
+endif()
```

Comment:
@dmakoviichuk-tt why is this a fatal error for building?

We don't normally set this when building tt-metal as it is a runtime requirement for us.